dags/oss_know/libs/util/base.py

```diff
@@ -52,3 +59,105 @@ def get_redis_client(redis_client_info):
     redis_client = redis.Redis(host=redis_client_info["HOST"], port=redis_client_info["PORT"], db=0,
                                decode_responses=True)
     return redis_client
+
+
+def infer_country_from_emailcctld(email):
+    """
+        :param  company: the company given by github
+        :return country_name  : the english name of a country
+        """
+    profile_domain = email.split(".")[-1].upper()
+    if profile_domain in CCTLD.keys():
+        return CCTLD[profile_domain]
+    else:
+        return None
+
+
+def infer_country_from_emaildomain(email):
+    """
+        :param  company: the company given by github
+        :return country_name  : the english name of a country
+        """
+    emaildomain = str(re.findall(r"@(.+?)\.", email))
+    if emaildomain in COMPANY_COUNTRY.keys():
```

Comment:
COMPANY_COUNTRY seems to be a 'big' object, checking existence of key in dict(which is a hash structure) should take much less time than checking item existence in list.

Reply:
Done